File: PopularShows/app/src/main/java/com/lgcode/popularshows/TVShowListActivity.java
package com.lgcode.popularshows;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.lgcode.popularshows.bus.BusProvider;
import com.lgcode.popularshows.models.TvShowListModel;
import com.lgcode.popularshows.presenters.TvShowListPresenter;
import com.lgcode.popularshows.views.TvShowListView;
import com.squareup.otto.Bus;

import glb.lg.popularshows.R;

public class TVShowListActivity extends AppCompatActivity {

    private Bus bus = BusProvider.getInstance();
    private TvShowListPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.show_list);

        this.presenter = new TvShowListPresenter(
                new TvShowListModel(bus),
                new TvShowListView(this, bus));
    }

    @Override
    protected void onStart() {
        super.onStart();
        bus.register(presenter);
    }

    @Override
    protected void onStop() {
        super.onStop();
        bus.unregister(presenter);
    }
}

File: KtPopularShows/app/src/main/java/example/lgcode/ktpopularshows/activity/TVShowListActivity.kt
package example.lgcode.ktpopularshows.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import dagger.android.AndroidInjection
import example.lgcode.ktpopularshows.R
import example.lgcode.ktpopularshows.mvp.presenter.TVShowListPresenter
import javax.inject.Inject

class TVShowListActivity: AppCompatActivity() {

    @Inject
    lateinit var presenter: TVShowListPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_list)
        //presenter = TVShowListPresenter(TVShowListModel(TVShowsRepository()), TVShowListView(this))
        presenter.setUp(this)
    }
}